File: leetcode/100-999/136. Single Number/solution_5.cs
public class Solution
{
    public int SingleNumber(int[] nums)
    {
        Dictionary<int, int> numsAndCount = new Dictionary<int, int>();

        foreach (int n in nums)
            if (numsAndCount.ContainsKey(n))
                numsAndCount[n]++;
            else
                numsAndCount[n] = 1;

        foreach (var pair in numsAndCount)
            if (pair.Value == 1)
                return pair.Key;

        return -1;
    }
}

File: leetcode/100-999/136. Single Number/solution_5.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var singleNumber = function(nums) 
{   
   const numsAndCount = new Map()

   for (const n of nums)
        if (numsAndCount.has(n))
            numsAndCount.set(n, numsAndCount.get(n) + 1)
        else
            numsAndCount.set(n, 1)

   for (const [key, value] of numsAndCount.entries())
        if (value == 1)
            return key

   return -1
};